lib/Runtime/Library/DateImplementation.cpp

```diff
@@ -1306,6 +1307,35 @@ namespace Js {
                     if (lwT >= 60)
                         goto LError;
                     lwTime += lwT;
+                    ss = (ch == ':' || ch == '.') ? (pch++, ssMillisecond) : ssNil;
+                    break;
+                }
+                case ssMillisecond:
+                {
+                    AssertMsg(isNextFieldDateNegativeVersion5 == false, "isNextFieldDateNegativeVersion5 == false");
+
+                    size_t numOfDigits = static_cast<size_t>(pch - pchBase);
```

Comment:
This is incorrect.
First of all, you can't detect the number of digits from `pch` since there can be ignored characters like 
`console.log(Date.parse("2011-11-08 19:48:43:1"));` needs to have the same result as `console.log(Date.parse("2011-11-08 19:48:43:1 "));`

If only 1 digit is present, then it is not a hundred, but a digit `19:48:43:1` has `1`ms not `100`  similarly `19:48:43:15` has `15`ms not `150`

You can rewrite this whole case with
```cpp
                case ssMillisecond:
                {
                    AssertMsg(isNextFieldDateNegativeVersion5 == false, "isNextFieldDateNegativeVersion5 == false");
                    // ignore any digits beyond the third also ignores leading zeros
                    lwMillisecond = lwT % 1000;
                    ss = ssNil;
                    break;
                }
```

Reply:
I'm pretty sure the `1` in `Date.parse("2011-11-08 19:48:43.1")` is 100 ms, not 1 ms. Both SpiderMonkey and v8 return 1320799723100. If it meant to be 1 ms, it should be 1320799723001.

Reply:
Interesting
`Date.parse("2011-11-08 19:48:43.1")` is indeed `1320810523100`
However, `Date.parse("2011-11-08 19:48:43:1")` is `1320810523001`
It seems, depending on wether we use `.` or `:` as a separator, it changes the meaning the next value.

Also, the following
```
console.log(Date.parse("2011-11-08 19:48:43.0010"));
console.log(Date.parse("2011-11-08 19:48:43:0010"));
```
is also very different.

It would seem that your implementation matches more the `.` pattern and mine the `:` pattern.
Therefore, we need to differenciate between the 2 when interpreting the value.

Note, that we still need a different mechanism than `static_cast<size_t>(pch - pchBase);` to determine the number of digits to take spaces into account.
We could probably just calculate it before `// skip to the next real character`

Reply:
Sorry I think I misunderstood your comment. Now I see `19:48:43.1` and `19:48:43:1` represent different time.

Reply:
I think I was also confused about that fact when I first wrote my original comment as I only tested with `:` initially